crates/oci-tar-builder/src/bin.rs

```diff
@@ -0,0 +1,101 @@
+use std::{env, fs};
+
+use {
+    anyhow::Context, clap::Parser, oci_spec::image as spec, oci_tar_builder::Builder,
+    std::fs::File, std::path::PathBuf,
+};
+
+pub fn main() {
+    let args = Args::parse();
+
+    let out_dir;
+    if let Some(out_path) = args.out_path.as_deref() {
+        out_dir = PathBuf::from(out_path);
+    } else {
+        out_dir = env::current_dir().unwrap();
+    }
+
+    let mut builder = Builder::default();
+
+    if let Some(module_path) = args.module.as_deref() {
+        let module_path = PathBuf::from(module_path);
+        builder.add_layer_with_media_type(
+            &module_path,
+            "application/vnd.w3c.wasm.module.v1+wasm".to_string(),
+        );
+    }
+
+    if let Some(components_path) = args.components.as_deref() {
+        let paths = fs::read_dir(components_path).unwrap();
+
+        for path in paths {
+            let path = path.unwrap().path();
+            let ext = path.extension().unwrap().to_str().unwrap();
+            match ext {
+                "wasm" => {
+                    builder.add_layer_with_media_type(
+                        &path,
+                        "application/vnd.wasm.component.v1+wasm".to_string(),
+                    );
+                }
+                "yml" => {
+                    builder.add_layer_with_media_type(
+                        &path,
+                        "application/vnd.wasm.component.config.v1+json".to_string(),
```

Comment:
Mapping `yml` files to `json` media type? It would work in the other direction because yaml is a superset of json.